plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/controller/Browse.js

```diff
@@ -0,0 +1,222 @@
+          updated: me.setAssetIcon
+        },
+        'nx-coreui-browsefeature nx-coreui-browse-result-list #filter': {
+          search: me.onSearch,
+          searchcleared: me.onSearchCleared
+        }
+      }
+    });
+  },
+
+  /**
+   * @override
+   */
+  getDescription: function(model) {
+    return model.get('name');
+  },
+
+  /**
+   * @override
+   * When a list managed by this controller is clicked, route the event to the proper handler
+   */
+  onSelection: function(list, model) {
+    var me = this,
+        modelType;
+
+    // Figure out what kind of list we’re dealing with
+    modelType = model.id.replace(/^.*?model\./, '').replace(/\-.*$/, '');
+
+    if (modelType == "Repository") {
+      me.onRepositorySelection(model);
+    }
+    else if (modelType == "Component") {
+      me.onComponentSelection(model);
+    }
+  },
+
+  /**
+   * @private
+   *
+   * Load browse results for selected repository.
+   *
+   * @param {NX.coreui.model.Repository} model selected repository
+   */
+  onRepositorySelection: function(model) {
+    var me = this,
+        browseResultStore = me.getBrowseResultStore(),
+        results = me.getResults(),
+        componentFilter = me.getComponentFilter();
+
+    browseResultStore.filters.removeAtKey('filter');
+    componentFilter.clearSearch();
+    results.getSelectionModel().deselectAll();
+    browseResultStore.addFilter([
+      {
+        id: 'repositoryName',
+        property: 'repositoryName',
+        value: model.get('name')
+      }
+    ]);
+  },
+
+  /**
+   * @private
+   *
+   * Show component details and load assets for selected component.
+   *
+   * @param {NX.coreui.model.Component} model selected component
+   */
+  onComponentSelection: function(model) {
+    var me = this;
+
+    me.getComponentDetails().setComponentModel(model);
+    me.getAssets().setComponentModel(model);
+  },
+
+  /**
+   * @private
+   * Filter grid.
+   */
+  onSearch: function(searchbox, value) {
+    var me = this,
+        browseResultStore = me.getBrowseResultStore(),
+        results = me.getResults(),
+        emptyText = NX.I18n.get('Coreui_View_Browse_BrowseResultList_EmptyText_Filter');
+
+    if (!browseResultStore.emptyText) {
+      browseResultStore.emptyText = results.getView().emptyText;
+    }
+    results.getView().emptyText = '<div class="x-grid-empty">' + emptyText.replace(/\$filter/, value) + '</div>';
+    results.getSelectionModel().deselectAll();
+    browseResultStore.addFilter([
+      {
+        id: 'filter',
+        property: 'filter',
+        value: value
+      }
+    ]);
+  },
+
+  /**
+   * @private
+   * Clear filtering on grid.
+   */
+  onSearchCleared: function() {
+    var me = this,
+        browseResultStore = me.getBrowseResultStore(),
+        results = me.getResults();
+
+    results.getView().emptyText = browseResultStore.emptyText;
+    results.getSelectionModel().deselectAll();
+    if (browseResultStore.filters.removeAtKey('filter')) {
```

Comment:
IIUC browseResultStore.removeFilter('filter') will apply the filter change and avoid the need for this check?